agenda/data/src/main/java/com/gkp/agenda/data/workers/CreatedItemsWorker.kt

```diff
@@ -35,23 +34,20 @@ class CreatedItemsWorker(
             return Result.failure()
         }
         val agendaItemId = inputData.getString(AGENDA_ITEM_ID) ?: return Result.failure()
-        println(" ATMS CreatedItemsWorker doWork agendaItemId $agendaItemId")
         return try {
-            val agendaItem = localDataSource.getAgendaItemById(agendaItemId)
-            performPushCreatedRequest(agendaItem)
+            val agendaItem = pendingSyncCreatedAgendaItemsDao.getCreatedAgendaItemById(agendaItemId)?: return Result.failure()
+            performPushCreatedRequest(agendaItem.agendaItem.toAgendaItem())
             pendingSyncCreatedAgendaItemsDao.deleteById(agendaItemId)
```

Comment:
Could it make more sense only to delete the sync item when the above request succeeds and return a retry result? Something like this perhaps?

```
return when(val result = remoteDataSource.postReminder(reminder)) {
  is com.plcoding.tasky.core.domain.util.Result.Error -> {
    result.error.toWorkerResult() // extension function to map the custom error type to either Workmanager’s Result.retry() or Result.failure()
  }
  is com.plcoding.tasky.core.domain.util.Result.Success -> {
    pendingSyncDao.deleteReminderPendingSyncEntity(pendingReminderId)
    Result.success()
  }
}
```